scalding-core/src/main/scala/com/twitter/scalding/Operations.scala

```diff
@@ -166,30 +156,93 @@ package com.twitter.scalding {
       }
     }
 
-    override def operate(flowProcess: FlowProcess[_], functionCall: FunctionCall[SummingCache[Tuple, V]]) {
+    override def operate(flowProcess: FlowProcess[_], functionCall: FunctionCall[MapsideCache[V]]) {
       val cache = functionCall.getContext
       val keyValueTE = functionCall.getArguments
       // Have to keep a copy of the key tuple because cascading will modify it
       val key = keyValueTE.selectEntry(keyFields).getTupleCopy
       val value = conv(keyValueTE.selectEntry(valueFields))
-      add(cache.put(Map(key -> value)), functionCall)
+      val (stats, evicted) = cache.put(key, value)
+      add(evicted, functionCall)
+
+      //iterator for performance
```

Comment:
Speaking of performance, since we're concerned to the point of wanting iterators instead of closures, using maps at all seems like it'd be slower than using objects here. I haven't done the benchmarking, but might it be worth switching to stats objects instead?